packages/web-runtime/src/components/Topbar/Notifications.vue

```diff
@@ -198,28 +193,34 @@ export default {
           data?.sort((a, b) => (new Date(b.datetime) as any) - (new Date(a.datetime) as any)) || []
       } catch (e) {
         console.error(e)
-      } finally {
-        if (unref(dropdownOpened)) {
-          setAdditionalData()
-        }
-        loading.value = false
       }
     }).restartable()
 
-    const deleteNotifications = async (ids: string[]) => {
-      loading.value = true
+    const deleteNotificationsTask = useTask(function* (signal, ids) {
       try {
-        await clientService.httpAuthenticated.delete(
+        yield clientService.httpAuthenticated.delete(
           'ocs/v2.php/apps/notifications/api/v1/notifications',
           { data: { ids } }
         )
       } catch (e) {
         console.error(e)
       } finally {
         notifications.value = unref(notifications).filter((n) => !ids.includes(n.notification_id))
-        loading.value = false
       }
-    }
+    }).restartable()
+
+    const computeNotificationDataTask = useTask(function* (signal, notifications) {
+      for (const notification of unref(notifications)) {
+        if (!notification.computedMessage) {
+          notification.computedMessage = yield new Promise((resolve) =>
+            resolve(getMessage(notification))
+          )
+        }
+        if (!notification.computedLink) {
+          notification.computedLink = yield new Promise((resolve) => resolve(getLink(notification)))
```

Comment:
same here

Reply:
sonarcloud is reporting a bug, if you don't use yield in a useTask generator func